x-pack/plugins/observability_solution/observability/public/pages/overview/components/sections/metrics/host_link.tsx

```diff
@@ -5,19 +5,40 @@
  * 2.0.
  */
 import React from 'react';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import {
+  ASSET_DETAILS_LOCATOR_ID,
+  AssetDetailsLocatorParams,
+} from '@kbn/observability-shared-plugin/common';
+import { SharePluginStart } from '@kbn/share-plugin/public';
 import { StringOrNull } from '../../../../..';
 
 interface Props {
   name: StringOrNull;
-  id: StringOrNull;
+  id: string;
   timerange: { from: number; to: number };
 }
 
 export function HostLink({ name, id, timerange }: Props) {
-  const link = `../../app/metrics/link-to/host-detail/${id}?from=${timerange.from}&to=${timerange.to}`;
+  const { services } = useKibana<{ share?: SharePluginStart }>();
+
+  const assetDetailsLocator =
+    services.share?.url.locators.get<AssetDetailsLocatorParams>(ASSET_DETAILS_LOCATOR_ID);
+
+  const href = assetDetailsLocator?.getRedirectUrl({
+    assetType: 'host',
+    assetId: id,
+    assetDetails: {
+      dateRange: {
+        from: new Date(timerange.from).toISOString(),
+        to: new Date(timerange.to).toISOString(),
+      },
+    },
+  });
+
   return (
     <>
```

Comment:
not part of this PR, but we don't need this fragment, do we?

Reply:
React needs a fragment or a wrapper when there are multiple children. It's not the case here. But since it's not part of this PR we can left it as-is.